crossdock/server/server.py

```diff
@@ -88,14 +95,46 @@ def tchannelNotImplResponse():
     return TraceResponse(notImplementedError="python <=> tchannel not implemented")
 
 
-@tchannel.thrift.register(service.TracedService)
-def startTrace(request):
-    return trace_response_to_thriftrw(service, tchannelNotImplResponse())
+class UnimplementedEndpointException(Exception):
+    pass
+
+
+def make_tchannel(port):
+    global tchannel
+    tchannel = TChannel('python', hostport="localhost:%d" % port, trace=True)
 
+    service = get_thrift_service(service_name='python')
 
-@tchannel.thrift.register(service.TracedService)
-def joinTrace(request):
-    return trace_response_to_thriftrw(service, tchannelNotImplResponse())
+    @tchannel.thrift.register(service.TracedService, method='startTrace')
+    @tornado.gen.coroutine
+    def start_trace(request):
+        raise UnimplementedEndpointException("tchannel startTrace endpoint not implemented")
```

Comment:
just don't add this endpoint to begin with, tchannel will throw some exception automatically. And we won't need UnimplementedEndpointException